### DISCUSSION TITLE
Light sleep mode and variables lifespan

### DISCUSSION THREAD

**Discussion by ErnestBrandi** [vercel/vercel#5206] [General] [answered]
Hi, I discovered about "light sleep" process when reading about "cron jobs" and other vercel's [issues](https://github.com/vercel/vercel/issues/189). I wonder what it does precisely; for example does restarting/awakening the instance recompile files ? 

I also wonder if it affects variables lifespan. In one project I sort of temporarily store api tokens in global variables (server side). If I come back to my website after a while, those variables seems undefined. 

Also if a variable is set (to a certain value) on a certain page, it seems to be reset (to its initial value) if I load another page (after "light sleep" happens). The value is then preserved if I keep browsing.

Finally, and if all of this is due to it, "light sleep" seems to happens quite fast (maybe after 5 - 10 minutes) ! Is this normal ?

Thanks for clarifications :)

**Comment by paulogdm** [comment, marked as answer]
Hello,

The issue that you linked is **very old**. It refers to Now 1.0, which doesn't exist anymore.

At the current model, deployments are immutable and you need to store state in a DB or storage. The Serverless model is also "on-demand", which means you can't expect any persistence between requests.

**Reply by ErnestBrandi** [reply]
Hi, thanks for answering. But is this sleep functionality still up to date ? If so, where can I find more infos about it ?
About the "on-demand", does this means if I want to access a variable from 3 different serverless functions, it will actually be different variables ? Those "functions" can't share any value otherwise than accessing them from a third party ?

**Reply by paulogdm** [reply]
The "sleep functionality" is simply not applicable here. When you hit a Serverless Function the request is "self-contained" and to share value between functions, you need a DB like REDIS or MongoDB.

You can read more about "Serverless Functions" if you search online for "AWS Lambda"

**Reply by ErnestBrandi** [reply]
For anyone interested, this is what put me on the wrong track (with serverless functions and global states) : https://blog.scottlogic.com/2019/03/04/lambda-global-state.html
Global variables can "leak" from one call to another (keep a previously set value for a random period of time).

**Reply by duhd1993** [reply]
But there is still sleeping, right? If your app is idle for some time, it needs to start and initialize. In the logs, there would be init duration, which could take a few seconds. If it's already initialized, this time is none. @paulogdm There is just nowhere to find how the sleeping is determined. Is it ok to use a cron bot to keep it alive?

**Reply by paulogdm** [reply]
Using a cronjob to keep the function up will likely cause a breach of the [Fair Use Policy](https://vercel.com/docs/platform/fair-use-policy). I don't recommend it.

Refer to https://vercel.com/docs/serverless-functions/conceptual-model for further info.
